### DISCUSSION TITLE
Producing TestNG XML reports with custom attributes in gradle

### DISCUSSION THREAD

**Discussion by itkhanz** [cbeust/testng#3128] [Q&A] [answered]
Hi,
I would like to have custom attributes in TestNG XML report. In the documentation, i read that this is possible via:
https://testng.org/#_xml_reports

but I could not figure out how to achieve this using gradle test task with useTestNG()

More detailed information on this I posted in gradle discsssion forms:
https://discuss.gradle.org/t/how-to-pass-custom-arguments-to-testng-listeners-in-gradle/48462

I researched about this issue alot, and came across some old issues and your response in testng user group where it was suggested to create a custom gradle task, but I am more interested to see whether we can achieve this via plain gradle test task.

Probably it is a gradle specific issue and limitation in gradle API to invoke TestNG, but I would still be interested to hear your opinion about any possibility.

**Comment by krmahadevan** [comment, marked as answer]
@itkhanz - The TestNG team does not own the Gradle integration piece. So you would need to check this with the Gradle community.

I think only the most common things are available out of the box via the `useTestNG {}` method parameter [TestNGOptions](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/testing/testng/TestNGOptions.html)

Since Gradle allows you to define your own custom task, you should perhaps pursue that.

**Reply by itkhanz** [reply]
Thank you for clarifying this issue. I have raised a [feature request](https://github.com/gradle/gradle/issues/29135) for Gradle Team to allow this option out of box via `useTestNG{} `inside gradle test task.

The problem with using custom gradle task of type `JavaExec` to achieve this functioanlity is that we then lose all the features provided by gradle that are part of task with type `Test`.
